app.rb

```diff
@@ -16,15 +59,280 @@ class MyApp < Sinatra::Base
     end
   end
 
-  error do |e|
-    if e.is_a? ActiveRecord::RecordNotFound
-      halt 404, json(error: "Not Found")
-    elsif e.is_a? ActiveRecord::RecordInvalid
-      halt 422, json(error: e.message)
+
+  def require_login!
+    unless current_user
+    redirect '/'
+    end
+  end
+
+  before do
+    unless ["/", "/newuser", "/dashboard"].include?(request.path)
+      require_login!
+      return
+    end
+  end
+
+  if ENV['PRY_ON_ERROR']
+    error do |e|
+      binding.pry
+    end
+  end
+
+  # login page show
+  get '/' do
+    erb :login
+  end
+
+
+#account page
+  get '/account' do
+    erb :account
+  end
+
+  #reset password
+  get '/password' do
+    erb :password
+  end
+
+  post '/password' do
+    if (current_user.password == Digest::SHA256.hexdigest(params[:oldpassword])) && (params[:newpassword1] == params[:newpassword2])
+      change_password(params[:newpassword1])
+      session[:message] = "Password Changed Successfully"
+      redirect '/account'
+    elsif current_user.password != Digest::SHA256.hexdigest(params[:oldpassword])
+      session[:message] = "Old Password Incorrect"
+      redirect '/password'
+    else
+      session[:message] = "New Password Did Not Match"
+      redirect '/password'
+    end
+  end
+
+# if login info is not found redirect to new user page
+  post '/' do
+    u = User.find_by(email: params[:username])
+    if u && u.password == Digest::SHA256.hexdigest(params[:password])
+      login_user u
+      redirect '/dashboard'
+    else
+      @failed_login = true
+      erb :login
+    end
+  end
+
+
+  post '/logout' do
+    logout
+    redirect '/'
+  end
+
+  # create new user info
+  #puts Digest::SHA256.hexdigest "Hello World"
+  post '/newuser' do
+    u = User.create!(email: params[:username], password: params[:password])
+    login_user u
+    session[:message] = "Account Creation Successful"
+    redirect '/dashboard'
+  end
+
+  # new user page show
+  get '/newuser' do
+    erb :newuser
+  end
+
+  # get songs info for dashboard
+  get '/dashboard' do
+    @songs = Playlist.current_alpha_hash
+    erb :dashboard
+  end
+
+  post '/invite' do
+    unless params[:email] == ""
+    Pony.mail :to => params[:email],
+              :from => "friend@alphatunez.herokuapp.com",
+              :headers => { 'Content-Type' => 'text/html' },
+              :subject => "Welcome to AlphatuneZ!",
+              :body => body = erb(:invite_email, layout: false )
+    end
+    session[:message] = "Email sent to #{params[:email]}"
```

Comment:
This *might* be a lie, if the email was blank. It'd be better to check that and re-display the invite form though, if that's the case.